src/compiler/sourcemap.ts

```diff
@@ -333,12 +336,21 @@ namespace ts {
                 const oldSource = currentSource;
                 if (source === oldSource) source = undefined;
 
+                let nameIndex;
+                if (isIdentifier(node)) {
+                    const originalName = getTextOfNodeFromSourceText(currentSourceText, node);
+                    nameIndex = sourceMapData.sourceMapNames.indexOf(originalName);

```

Comment:
For larger files, as the array increases in size it will take longer and longer to find names. I would recommend using a `ts.Map` (via a call to `createMap`) to store the name index for each recorded name.